be/src/common/exception_stack_trace.h

```
@@ -0,0 +1,238 @@
+            //vDynObjectFileName = dwfl_module_info(module, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+            vDWARFObjLine = dwfl_getsrc(dwfl, addr);
+            if (vDWARFObjLine != NULL) {
+                vSourceFileName = dwfl_lineinfo(vDWARFObjLine, &addr, &vSourceFileLineNumber, NULL, NULL, NULL);
+                //fprintf(stderr, " %s:%d", strrchr(vSourceFileName, '/')+1, vSourceFileLineNumber);
+            }
+        }
+        if (dwfl == NULL || vDWARFObjLine == NULL || vSourceFileName == NULL) {
+            vSourceFileName = "???";
+            vSourceFileLineNumber = 0;
+        }
+
+        // Print the stack frame number:
+        fprintf(stderr, "#%2d:", ++n);
+
+        // Print the stack addresses:
+        fprintf(stderr, " 0x%016" PRIxPTR " sp=0x%016" PRIxPTR, static_cast<uintptr_t>(ip), static_cast<uintptr_t>(sp));
+
+        // Print the source file name:
+        fprintf(stderr, " %s:%d", vSourceFileName, vSourceFileLineNumber);
+
+        // Print the dynamic object file name (that is the library name).
+        // This is typically not interesting if we have the source file name.
+        //fprintf(stderr, " %s", vDynObjectFileName);
+
+        // Print the procedure name:
+        fprintf(stderr, " %s", vProcedureName);
+
+        // Print the procedure offset:
+        //fprintf(stderr, " + 0x%" PRIxPTR, static_cast<uintptr_t>(off));
+
+        // Print a newline to terminate the output:
+        fprintf(stderr, "\n");
+
+        // Stop the stack trace at the main method (there are some
+        // uninteresting higher level functions on the stack):
+        if (strcmp(vProcedureName, "main") == 0) {
+            break;
+        }
+
+        vUnwindStatus = unw_step(&vUnwindCursor);
+        if (vUnwindStatus < 0) {
+            fprintf(stderr, "libunwind_print_backtrace(): Error in unw_step: %d\n", vUnwindStatus);
+        }
+    }
+}
+
+void __wrap___cxa_throw(void* thrown_exception, void* infov, void (*dest)(void*)) {
```

`__wrap___cxa_throw ` will be called in which situation? can it catch all exceptions?

---

`__cxa_throw` is a libc function to process throwing exceptions. `__wrap___cxa_throw`  is to rewrite it. 
yes in my understanding.